src/ceph-daemon

```diff
@@ -0,0 +1,1289 @@
+
+        # tmp keyring file
+        tmp_keyring = tempfile.NamedTemporaryFile(mode='w')
+        os.fchmod(tmp_keyring.fileno(), 0o600)
+        tmp_keyring.write(keyring)
+        tmp_keyring.flush()
+
+        # tmp config file
+        tmp_config = tempfile.NamedTemporaryFile(mode='w')
+        os.fchmod(tmp_config.fileno(), 0o600)
+        tmp_config.write(config)
+        tmp_config.flush()
+
+        mounts[tmp_config.name] = '/etc/ceph/ceph.conf:z'
+        mounts[tmp_keyring.name] = '/var/lib/ceph/bootstrap-osd/ceph.keyring:z'
+
+    c = CephContainer(
+        image=args.image,
+        entrypoint='/usr/sbin/ceph-volume',
+        args=args.command,
+        podman_args=['--privileged'],
+        volume_mounts=mounts,
+    )
+    subprocess.call(c.run_cmd())
+
+##################################
+
+def command_unit():
+    (daemon_type, daemon_id) = args.name.split('.')
+    c = get_container(args.fsid, daemon_type, daemon_id)
+    unit_name = get_unit_name(args.fsid, daemon_type, daemon_id)
+    subprocess.call([
+        'systemctl',
+        args.command,
+        unit_name])
+
+##################################
+
+def command_ls():
+    ls = []
+
+    # /var/lib/ceph
+    if os.path.exists(args.data_dir):
+        for i in os.listdir(args.data_dir):
+            if i in ['mon', 'osd', 'mds', 'mgr']:
+                daemon_type = i
+                for j in os.listdir(os.path.join(args.data_dir, i)):
+                    if '-' not in j:
+                        continue
+                    (cluster, daemon_id) = j.split('-', 1)
+                    fsid = get_legacy_daemon_fsid(cluster, daemon_type,
+                                                  daemon_id) or 'unknown'
+                    (enabled, active) = check_unit('ceph-%s@%s' % (daemon_type,
+                                                                   daemon_id))
+                    ls.append({
+                        'style': 'legacy',
+                        'name': '%s.%s' % (daemon_type, daemon_id),
+                        'fsid': fsid,
+                        'enabled': enabled,
+                        'active': active,
+                    })
+            elif is_fsid(i):
+                fsid = i
+                for j in os.listdir(os.path.join(args.data_dir, i)):
+                    (daemon_type, daemon_id) = j.split('.', 1)
+                    (enabled, active) = check_unit(get_unit_name(fsid,
+                                                                 daemon_type,
+                                                                 daemon_id))
+                    ls.append({
+                        'style': 'ceph-daemon:v1',
+                        'name': '%s.%s' % (daemon_type, daemon_id),
+                        'fsid': fsid,
+                        'enabled': enabled,
+                        'active': active,
+                    })
+
+    # /var/lib/rook
+    # WRITE ME
+
+    print(json.dumps(ls, indent=4))
```

Comment:
if the data_dir doesn't exist this will get executed and break

Reply:
ls will be []